src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -414,6 +417,51 @@ public Void clearEntitySet( UUID entitySetId ) {
         return null;
     }
 
+    @Override
+    @RequestMapping(
+            path = { "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + ENTITY_KEY_ID_PATH + "/graph" },
+            method = RequestMethod.DELETE
+    )
+    public Void clearEntityAndNeighborEntities(
+            @PathVariable( ENTITY_SET_ID ) UUID entitySetId,
+            @PathVariable( ENTITY_ID ) UUID entityId
+    ) {
+
+        ensureReadAccess( new AclKey( entitySetId ) );
+        Map<UUID, List<NeighborEntityDetails>> result = searchService
+                .executeEntityNeighborSearch( entitySetId, ImmutableSet.of( entityId ) );
+
+        if ( result != null && !result.containsKey( entityId ) ) {
```

Comment:
This should only continue if the result does contain the key, right?

Reply:
Good catch.